### PR TITLE
Fix build with newer version of spdlog

### PR DESCRIPTION
**Pull request by adam4813**
Newest protobuf needed some warning fixes as well.

### CODE REVIEW

File: client/CMakeLists.txt [hunk 1]
```diff
@@ -58,6 +51,10 @@ set(trillek-client_SOURCES ${trillek-client_SOURCES} PARENT_SCOPE) # so tests ca
 set(OSX_LIBRARIES "")
 if (APPLE)
 	set(OSX_LIBRARIES "/usr/lib/libobjc.dylib")
+	
+	# Rather wonderfully, Apple decided to mark every single fucking OpenGL function
+	# as deprecated because they have 'deprecated' OpenGL. Brilliant!
+    add_definitions(-DGL_SILENCE_DEPRECATION)
```
Comment (milesrout) [inline client/CMakeLists.txt]: I like that you kept the comment intact XD

File: client/sound-system.hpp [hunk 2]
```diff
@@ -7,6 +7,7 @@
 #include <memory>
 #include <set>
 #include <iostream>
+#include <thread>
```
Comment (milesrout) [inline client/sound-system.hpp]: Is this necessary?

Comment (adam4813) [reply client/sound-system.hpp]: There is a call to `std::this_thread` in the update loop to sleep, so yes. Also the build fails without it now for some reason that it worked before 😞

File: client/CMakeLists.txt (hunk 1 above)
Comment (adam4813) [reply client/CMakeLists.txt]: Well comments like that are important, so we don't scratch our heads later like why did we need this define?

**Pull request merged**